File: src/workflow_shell/commands/git/fix_ignore.clj
(ns workflow-shell.commands.git.fix_ignore
  (:require [clojure.java.shell :refer [sh]])
  (:require [workflow-shell.commands.git.commit_all_message :refer [gcam]])
  (:gen-class))

(def gfi (hash-map
         :description "Attempts to fix gitignore not working properly"
         :execute (fn [& args]
                    (let [removal (sh "git" "rm" "-r" "--cached" ".")
                          addition (sh "git" "add" ".")]
                      (if-not (empty? (:out removal))
                        (println (:out removal))
                        ((println (:err removal))
                         (System/exit 0)))
                      (if-not (empty? (:out addition))
                        (println (:out addition))
                        ((println (:err addition))
                         (System/exit 0)))
                      ((:execute gcam) "Fixed gitignore")))))

File: src/commands/git/FixIgnore.ts
import { Option } from '../../common/interface/Option';
import { SuperCommand } from '../../common/SuperCommand';
import { executeCommand } from '../../common/Cli';
import { Command as GitCommit } from './CommitAllWithMessage';
import { Command as GitAddDot } from './AddDot';
import { Argument } from '../../common/interface/Argument';

const argumentss: Argument[] = [];
const options: Option[] = [];

export class Command extends SuperCommand {
  description: string = 'Attempts to fix gitignore not working properly.';
  alias: string = 'gfi';
  public async execute(vital?: boolean, input?: any) {
    await super.executeWithInput(argumentss, options, input, vital, async () => {
      await (new GitCommit()).execute(false, { message: 'Before Fixing gitignore' });
      await executeCommand('git', ['rm', '-r', '--cached', '.']);
      await (new GitAddDot()).execute(false);
      await (new GitCommit()).execute(false, { message: 'Fixed gitignore' });
    });
  }
}
